src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,77 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> linesOfCsvFile = getDataFromCsvfile(fromFileName);
+        int sumOfSupply = Constants.AMOUNT_STARTING_VALUE;
+        int sumOfBuy = Constants.AMOUNT_STARTING_VALUE;
+        if (!linesOfCsvFile.isEmpty()) {
+            String currentType = "";
+            int currentValue = 0;
+            for (int i = 0; i < linesOfCsvFile.toArray().length; i++) {
+                String[] itemDataSplit = linesOfCsvFile.get(i).split(Constants.COMMA_SEPARATOR);
+                if (itemDataSplit.length == 2) {
+                    currentType = itemDataSplit[Constants.ITEM_TYPE_INDEX];
+                    try {
+                        currentValue = Integer.parseInt(itemDataSplit[Constants.ITEM_VALUE_INDEX]);
+                    } catch (NumberFormatException e) {
+                        throw new RuntimeException(Constants.PARSE_INTEGER
+                                + fromFileName + " for line: '" + i + "'", e);
+                    }
```

After splitting the CSV line you do not `trim()` parsed tokens and you silently ignore lines that do not have exactly two tokens. This violates the requirement: "Extra whitespace around fields: trim parsed tokens before processing to ensure correctness" and also violates: "Invalid integer in amount or malformed line (wrong number of tokens): should cause a handled exception path that results in throwing a RuntimeException with an informative message (see checklist item “Don't ignore exceptions")." 

Fixes: - After `split`, call `trim()` on both tokens before comparing/parsing so inputs like `supply, 30` are handled. - If `parts.length != 2`, throw a RuntimeException that includes the filename and the line index/content (do not silently skip).